mdsuite/visualizer/d2_data_visualization.py

```diff
@@ -40,19 +40,21 @@ class DataVisualizer2D:
     Visualizer for two-dimensional data.
     """
 
-    def __init__(self, title: str):
+    def __init__(self, title: str, path: str):
         """
         Constructor for the data visualizer.
 
         Parameters
         ----------
         title : str
                 title of the plot.
+        path : str
+                path to the saving directory of the plot
         """
         if config.jupyter:
             output_notebook()
         else:
-            output_file(f"{title}.html", title=title)
+            output_file(f"{path}/{title}.html", title=title)
```

Comment:
As mentioned in my other comment, path is a `pathlib` object, so consider using `(path / title).with_suffix(".html")` or `f"{path / title}.html"` or  `(path / title).with_suffix(".html").as_posix()` if above fails. You can not hard code the `/` because we need to be os independent.

Reply:
Yep this needs changing.